src/EditorFeatures/CSharpTest/UseRecursivePatterns/UseRecursivePatternsTests.cs

```diff
@@ -0,0 +1,239 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.UseRecursivePatterns;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.UseRecursivePatterns
+{
+    public partial class UseRecursivePatternsTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+           => new CSharpUseRecursivePatternsCodeRefactoringProvider();
+
+        [Fact]
+        public async Task Test1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return b.P1 == 1 [||]&& b.P2 == 2;
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return b is
+        {
+            P1: 1, P2: 2
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return b.MetadataName == string.Empty &&
+            b.ContainingType == null &&
+            b.ContainingNamespace != null &&
+            b.ContainingNamespace.Name == string.Empty [||]&&
+            b.ContainingNamespace.ContainingNamespace == null;
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return b is
+        {
+            MetadataName: string.Empty, ContainingType: null, ContainingNamespace:
+            {
+                Name: string.Empty, ContainingNamespace: null
+            }
+        };
```

Comment:
the formatting here seems wonky.  where it doesn't seem to match up with original like i woudl expect.

Reply:
here's my thinking.  if the original was single-line, we should attempt to preserve single-line.  If the original is multi-line, we can be multiline too.  When multi-line, we should keep properties on a single line, if their *patterns* are single line.  But if they are multi-line, then they should go across multiple lines as well.  So this should be:

```c#
        return b is
        {
            MetadataName: string.Empty,
            ContainingType: null,
            ContainingNamespace:
            {
                Name: string.Empty, ContainingNamespace: null
            }
        };
```